Sources/LanguageServerProtocol/SupportTypes/WorkspaceFolder.swift

```diff
@@ -18,13 +18,23 @@ public enum BuildConfiguration: Hashable, Codable {
   case release
 }
 
+/// The type of workspace; default workspace type selection logic can be overridden.
+///
+/// **(LSP Extension)**
+public enum WorkspaceType: Hashable, Codable {
+  case buildserver, compdb, swiftpm
```

Comment:
Nitpick, let’s go for more descriptive names.
```suggestion
  case buildServer, compilationDatabase, swiftPM
```